numba_dppy/utils/array_utils.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import dpctl
+import dpctl.memory as dpctl_mem
+
+supported_numpy_dtype = [
+    np.int32,
+    np.int64,
+    np.uint32,
+    np.int64,
+    np.float32,
+    np.float64,
+]
+
+
+def is_usm_backed(obj):
+    """
+    Determine and return a SYCL device accesible object.
+
```

Comment:
Detailed description is required. Especially what does `dpctl.memory.as_usm_memory` does should be summarized here.